core/trino-main/src/main/java/io/trino/operator/DynamicFilterSourceOperator.java

```diff
@@ -487,4 +487,106 @@ public boolean isFinished()
             return current == null && finished;
         }
     }
+
+    // Simple object to encapsulate whether channel is handled by set, range or neither
+    private class ChannelsState
```

Comment:
Can we instead make a per channel `ChannelFilter` class and make the processing of block and fallback logic an internal detail of this class ?